### PR TITLE
Host Maintenance Policy implementation

### PR DESCRIPTION
**Pull request by kfswain**
Adding HostMaintenancePolicy support for GKE terraform

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

```release-note:enhancement
container: added `node_config.host_maintenance_policy` field to `google_container_cluster` and `google_container_node_pool`
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb [hunk 1]
```diff
@@ -3285,4 +3287,50 @@ resource "google_container_node_pool" "with_tpu_topology" {
...
+}
+
+resource "google_container_node_pool" "np" {
+  name               = "%s"
+  location           = "us-central1-a"
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 1
+  node_config {
+	machine_type = "n2-standard-2"
+    host_maintenance_policy {
```
Comment (trodge) [inline mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:3329]: Could you fix the indentation here?

**Review by trodge — CHANGES_REQUESTED** [review]
Please add a test for the field in the `google_container_cluster` resource as well.

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb (hunk 1 above)
Comment (kfswain) [reply mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:3334]: Done. Interesting, looks like tab is interpreted as 8 spaces in this editor. Swapped to spaces.

**Comment by kfswain** [comment]
In the process of getting the terraform test GCP projects allowlisted into this feature. It is very limited availability, but I have ran these tests manually and confirmed it works.

**Comment by kfswain** [comment]
> Please add a test for the field in the `google_container_cluster` resource as well.

Done

**Review by trodge — APPROVED** [review]

**Pull request merged**